library/Booster/Pattern/ApplyEquations.hs

```diff
@@ -745,8 +745,6 @@ simplifyConstraint' recurseIntoEvalBool = \case
         | isConstructorSymbol symbol = pure FalseBool
     evalEqualsK KSet{} (SymbolApplication symbol _ _)
         | isConstructorSymbol symbol = pure FalseBool
-    evalEqualsK _ (SymbolApplication symbol _ _)
-        | isConstructorSymbol symbol = pure FalseBool
     evalEqualsK (Injection s1L s2L l) (Injection s1R s2R r)
         | s1L == s1R && s2L == s2R = evalEqualsK l r
```

Comment:
This seems too restrictive, but maybe that's OK because it just won't handle some cases?

What about `inj(A, B){inj(B, C){term}}` and `inj(A, C){term}`?